### DISCUSSION TITLE
One of the constraints evaluates to a FALSE LOGICAL variable. Your model is infeasible

### DISCUSSION THREAD

**Discussion by Yikxinxin** [yalmip/YALMIP#1327] [Optimization modelling] [answered]
Dear Johan, I encountered this error prompt while editing constraint conditions. May I know the reason? How can I solve this problem? Among them, delta (t)+p_ Fut (t) is a variable of type sdpvar.
![SharedScreenshot](https://github.com/yalmip/YALMIP/assets/104471890/27cd6721-b55a-4cd2-b36b-82d398957d71)
![2](https://github.com/yalmip/YALMIP/assets/104471890/0cc9e36b-89ec-4236-84d3-8ccdc606178a)

**Comment by johanlofberg** [comment, marked as answer]
Something evaluates to something like 1==0 

>> sdpvar x
>> y = x+1
>> x >= y
Error using  >= 
Error using constraint
Inequality constraint evaluated to trivial false (no decision variable in constraint)


Also, the use of short-circuit && appears odd since it should not work
>> x>=1 && x<=2
The following error occurred converting from constraint to logical:
Conversion to logical from constraint is not possible.

as you must use &
>> x>=1 & x<=2
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
|   ID|                    Constraint|   Coefficient range|
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
|   #1|   Element-wise inequality 1x1|              1 to 1|
|   #2|   Element-wise inequality 1x1|              1 to 2|
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Better to not use & as it is confusing to most, better to just concatenate

**Reply by johanlofberg** [reply]
actually, that must mean the stuff you have with p_Tra >= ... actually doesn't involve any sdpvars and evaluates to false

**Comment by Yikxinxin** [comment]
That's right, I got it. Thank you very much for pointing out my mistake!











在 2023-09-03 20:59:58，"Johan Löfberg" ***@***.***> 写道：

actually, that must mean the stuff you have with p_Tra >= ... actually doesn't involve any sdpvars and evaluates to false
